### ISSUE TITLE
Packet length < 30 bytes (bad detection of madwifi-ng)

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#34]
**Reported by misterx on 29 Dec 2006 18:05 UTC**

It seems like the packets being returned are 26 bytes in length.  That is: frame control (2 bytes), duration id (2 bytes), 3 mac addresses (18 bytes), CRC (4 bytes). In other words, there is no actual data being returned. This is happening on the authentication request.

I am not sure if this is the fault of aireplay-ng when it sends the authentication request or reads the response, the fault of the madwifi driver when it sends or receives, or due to the specifics of the routers. It doesn't seem to happen with all routers, but most.

This is the extent to which I can help with this I think.  Hopefully someone who can do more finds it useful.  Below are a couple of packets in hex.

```
B0 - 08 - 3A - 01 - 00 - 11 - 22 - 33 - 44 - 55 - 00 - 0B - 23 - EB - D4 - F2 - 00 - 0B - 23 - EB - D4 - F2 - 80 - 07 - 00 - 00
B0 - 08 - 3A - 01 - 00 - 11 - 22 - 33 - 44 - 55 - 00 - 0B - 23 - EB - D4 - F2 - 00 - 0B - 23 - EB - D4 - F2 - 90 - 07 - 00 - 00
B0 - 08 - 3A - 01 - 00 - 13 - 02 - A4 - 21 - 95 - 00 - 13 - 46 - FC - 80 - BC - 00 - 13 - 46 - FC - 80 - BC - 70 - 39 - 00 - 00
B0 - 00 - 3A - 01 - 00 - 13 - 02 - A4 - 21 - 95 - 00 - 13 - 46 - FC - 80 - BC - 00 - 13 - 46 - FC - 80 - BC - 90 - 39 - 00 - 00

```

(Taken from the forum)

**Comment by aircrack-ng** [comment]
**Comment by hirte on 29 Dec 2006 18:05 UTC**

First to the captured packet: that is not the crc value at the end, that's the squence/fragment counter (2 bytes) + authentication algorithm open system (2 bytes). So the packet is correct as far as we ignore the missing 4 bytes at the end for authentication sequence number (2 bytes) and status code (2 bytes). There are 4 bytes missing and we're working with atheros. Looking at read_packet() tells us, that we cut the last 4 bytes in case we're using madwifi but not madwifi-ng. So there could be an error in detecting madwifi-ng properly.
Maybe the svn version of madwifi-ng supports "rawdev" as a valid key, thus making our detection to produce false positives.
If this is not the problem and we're detecting the driver as madwifi-ng then i assume there is a bug in the driver itself.

See #54 for a quick debug output.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 29 Dec 2006 18:05 UTC**

On airodump-ng, the detection is ok.

The madwifi detection code in aireplay-ng (for this part) is not the same as airodump-ng (in this one, detection is ok).

If I'm not too tired when I come back home today, I'll commit the changes (I'm abroad on a hotspot and I don't like to put my login/pass on open networks). If it's not today it will be done next year (tomorrow ;)).

**Comment by aircrack-ng** [comment]
**Modified by misterx on 29 Dec 2006 18:05 UTC**

**Comment by aircrack-ng** [comment]
**Comment by misterx on 29 Dec 2006 18:05 UTC**

(In #58) Fixes bad detection of madwifi-ng in aireplay-ng.c (Closes: #34)

**Comment by aircrack-ng** [comment]
**Comment by misterx on 29 Dec 2006 18:05 UTC**

It seems to happen also with aircrack-ng/aircrack-ng@7fafac86c9639d613f20a5b6182bb2bf27de754d (bad madwifi-ng detection). Using interface 'ath22'. I'll put more debug information to see where it fails.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 29 Dec 2006 18:05 UTC**

(In #61) Workaround for madwifi interfaces name. They can be up to ath99; that should be enough for one computer (Closes: #34)
